rbac/api/cross_access/view.py

```diff
@@ -98,6 +98,8 @@ class CrossAccountRequestViewSet(
 
     def get_queryset(self):
         """Get query set based on the queryBy key word."""
+        if self.request.method in ["PATCH", "PUT"]:
```

Comment:
Just curious why this is required now?

Reply:
Yea, it was missing. The request will actually go to the validate_and_get_key, which is for the list, then because there is no query_by, it would set query_by to account and check if the this api call is made by admin. So we were using the wrong one,